pkg/ingester/ingester_v2.go

```diff
@@ -1611,6 +1617,10 @@ func (i *Ingester) compactionLoop(ctx context.Context) error {
 	ticker := time.NewTicker(i.cfg.BlocksStorageConfig.TSDB.HeadCompactionInterval)
 	defer ticker.Stop()
 
+	// Apply positive jitter only to ensure that the minimum timeout is adhered to.
+	i.TSDBState.compactionIdleTimeout = util.DurationWithPositiveJitter(i.cfg.BlocksStorageConfig.TSDB.HeadCompactionIdleTimeout, compactionIdleTimeoutJitter)
```

Comment:
Up for discussion here: Should we change the timeout e.g. each time the timeout is hit? This would mean that if each ingester picked similar timeouts, then we we would diverge next time around. The tricky bit here, is that we would need a timeout per-user, as each user can hit the timeout at different times (depending on `lastUpdateTime`).

Reply:
I would start simple setting it at startup, as you did.